### PR TITLE
Fibonacci

### PR DESCRIPTION
**Pull request by pwiatr3**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @pwiatr3

**Comment by ziobron** [comment]
⏱ 3 XP extra for @pwiatr3 for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -2,10 +2,37 @@
 
 int fibonacci_iterative(int sequence) {
     // TODO: Your implementation goes here
+        int an1 = 1; // a(n-1) - n-1 element of fibonacci series
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:5]: Wrong indentation

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -2,10 +2,37 @@
...
+    int an2 = 0; // a(n-2) - n-2 element of fibonacci series
+    int an = 0; // a(n) - n element of fibonacci series
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:7]: What is `an` ? Variable should always represent real operation.
Like `result` `divider` `multiplier` `error_code` etc...

File: homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -2,10 +2,37 @@
...
+
+    if(sequence == 0){
+        return 0;
+    }
+    if(sequence == 1){
+        return 1;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:14]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -2,10 +2,37 @@
...
 }
 
 int fibonacci_recursive(int sequence) {
     // TODO: Your implementation goes here
+        if(sequence == 0){
+        return 0;
+    }
+    if(sequence == 1){
+        return 1;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:33]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 5]
```diff
@@ -2,10 +2,37 @@
...
+        }
+        return an;
+    }
     return 0;
 }
 
 int fibonacci_recursive(int sequence) {
     // TODO: Your implementation goes here
+        if(sequence == 0){
+        return 0;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:29]: Wrong indentation

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**